### PR TITLE
Add exclude to options which allow to ignore operator groups.

### PR DESCRIPTION
**Pull request by lakerka**

### CODE REVIEW

File: src/binary/index.js [hunk 1]
```diff
@@ -1,15 +1,38 @@
...
+  return options.exclude.indexOf(ASSERTION_TO_EXCLUDE_OPTION_NAME[Assertion.name]) >= 0;
+}
+
+
 const assertions = [
-  require('./equal'),
-  require('./numeric'),
-  require('./add'),
-  require('./relational'),
+  equal,
```
Comment (vitalets) [inline src/binary/index.js]: I suggest to convert `assertions` into object:
```js
const assertions = [
  equal: require('./equal'),
  numeric : require('./numeric'),
  add: require('./add'),
  relational: require('./relational'),
];
```
This avoids adding constant `ASSERTION_TO_EXCLUDE_OPTION_NAME`.

File: src/options.js [hunk 2]
```diff
@@ -16,17 +18,14 @@ const defaults = {
   implicitEqualCustomTypes: 'deny',
   explicitEqualTrue: 'deny',
   explicitEqualFalse: 'deny',
+  exclude: []
```
Comment (vitalets) [inline src/options.js]: Let's name it more specifically: `excludeOperators`

File: test/specs/equal.spec.js [hunk 3]
```diff
@@ -147,7 +147,12 @@ describe('equal', function () {
     });
   });
 
+  describe('exclude = ["equal"]', function () {
```
Comment (vitalets) [inline test/specs/equal.spec.js]: I think test of this option should be in `options.spec` as it relates to all checks, not only equal. 
The test itself should check 'x === y' under different option values: 
* exclude: ['equal']
* exclude: ['add']
* exclude: ['add', 'equal']
* exclude: []
* exclude: ['foo']

File: src/binary/index.js [hunk 4]
```diff
@@ -1,15 +1,26 @@
...
+}
+
 
 module.exports = class BinaryAssertions {
   constructor(options) {
-    this._assertions = options.enabled ? assertions.map(Assertion => new Assertion(options)) : [];
+    this._assertions = [];
+    for (const operator of Object.keys(assertions)) {
+      const Assertion = assertions[operator];
+      if (options.enabled && !isExcluded(options, operator)) {
```
Comment (vitalets) [inline src/binary/index.js]: I think here we can optimize and move `options.enabled` check out of loop. And move loop itself into separate method. Also we can put hanging `isExcluded` into class.
E.g.
```js
module.exports = class BinaryAssertions {
  constructor(options) {
    this._assertions = [];
    this._options = options;
    if (this._options.enabled) {
      this._fillAssertions();
    }
  }
  _fillAssertions() {
    for (const operator of Object.keys(assertions)) {
      if (!this._isExcludedOperator(operator)) {
        const Assertion = assertions[operator];
        this._assertions.push(new Assertion(this._options));
      }
    }
  }
  _isExcludedOperator(operator) {
    return this._options.excludeOperators.indexOf(operator) >= 0;
  }
```

File: test/specs/options.spec.js [hunk 5]
```diff
@@ -8,6 +8,8 @@
 describe('options', function () {
 
   let f, spy;
+  const equal_tempale = 'Strict equal of different types: {x} === {y}';
```
Comment (vitalets) [inline test/specs/options.spec.js]: `const equalTempale`

File: test/specs/options.spec.js [hunk 6]
```diff
@@ -80,4 +82,71 @@ describe('options', function () {
     });
   });
 
+  describe('excludeOperators = ["equal"]', function () {
+    before(() => f = getFn('x === y',  {
+      excludeOperators: ['equal']
+    }));
+    it('does not warn for (null, number)', () => doesNotWarn(f, null, 1));
```
Comment (vitalets) [inline test/specs/options.spec.js]: Here (and below) it's enough to check that it does not warn for one combination of types. like you check in `mixed excludeOperators` section.

**Comment by vitalets** [comment]
Looks nice. I will check why travis reports failed build.

File: test/specs/options.spec.js [hunk 7]
```diff
@@ -8,6 +8,8 @@
 describe('options', function () {
 
   let f, spy;
+  const equalTempale = 'Strict equal of different types: {x} === {y}';
```
Comment (vitalets) [inline test/specs/options.spec.js]: `equalTemplate`

**Comment by lakerka** [comment]
Rebased.

**Comment by lakerka** [comment]
Do I need to do anything else?

**Comment by lakerka** [comment]
I think @vitalets you have to perform merge since I can see only "Close pull request" and "Comment" buttons.

**Comment by vitalets** [comment]
> I think @vitalets you have to perform merge since I can see only "Close pull request" and "Comment" buttons.

Yes. I will merge and release new version. 
Thanks for your support! 👍

**Pull request merged**